src/main/java/org/springframework/data/repository/core/support/MethodLookups.java

```diff
@@ -378,13 +377,17 @@ private static boolean parametersMatch(Method baseClassMethod, Method declaredMe
 					&& parameterCriteria.getBaseType().isAssignableFrom(parameterCriteria.getDeclaredType());
 		}
 
-		private static Stream<ParameterOverrideCriteria> methodParameters(Method first, Method second) {
+		private static boolean parameterCountMatch(InvokedMethod invokedMethod, Method baseClassMethod) {
 
-			Assert.isTrue(first.getParameterCount() == second.getParameterCount(), "Method parameter count must be equal!");
+			ImplementationInvocationMetadata invocationMetadata = new ImplementationInvocationMetadata(
```

Comment:
Does it make sense to cache those instances?

Reply:
In this area, we don't have a good place to host a cache. `RepositoryComposition` uses this method via `RepositoryFragments.findMethod(…)` and the outcome of `findMethod(…)` gets cached anyway so we should be good.